dev/manual_tests/lib/menu_bar.dart

```diff
@@ -0,0 +1,379 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/material.dart';
+import 'package:flutter/services.dart';
+
+enum TestMenu {
+  mainMenu1('Menu 1'),
+  mainMenu2('Menu 2'),
+  mainMenu3('Menu 3'),
+  mainMenu4('Menu 4'),
+  subMenu1('Sub Menu 1'),
+  subMenu2('Sub Menu 2'),
+  subMenu3('Sub Menu 3'),
+  subMenu4('Sub Menu 4'),
+  subMenu5('Sub Menu 5'),
+  subMenu6('Sub Menu 6'),
+  subMenu7('Sub Menu 7'),
+  subMenu8('Sub Menu 8'),
+  subSubMenu1('Sub Sub Menu 1'),
+  subSubMenu2('Sub Sub Menu 2'),
+  subSubMenu3('Sub Sub Menu 3');
+
+  const TestMenu(this.label);
+  final String label;
+}
+
+void main() {
+  debugFocusChanges = false;
+  runApp(const MaterialApp(
+    title: 'Menu Tester',
+    home: Material(child: Home()),
+  ));
+}
+
+class Home extends StatefulWidget {
+  const Home({super.key});
+
+  @override
+  State<Home> createState() => _HomeState();
+}
+
+class _HomeState extends State<Home> {
+  late MenuBarController controller;
+  bool isPlatformMenu = false;
+  VisualDensity density = VisualDensity.standard;
+  TextDirection textDirection = TextDirection.ltr;
+  bool enabled = true;
+  bool addItem = false;
+  bool checked = false;
+
+  void _itemSelected(TestMenu item) {
+    debugPrint('App: Selected item ${item.label}');
+  }
+
+  void _openItem(TestMenu item) {
+    debugPrint('App: Opened item ${item.label}');
+  }
+
+  void _closeItem(TestMenu item) {
+    debugPrint('App: Closed item ${item.label}');
+  }
+
+  @override
+  void initState() {
+    super.initState();
+    controller = MenuBarController();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    final ThemeData theme = Theme.of(context);
+    return Directionality(
+      textDirection: textDirection,
+      child: Builder(builder: (BuildContext context) {
+        return Theme(
+          data: theme.copyWith(visualDensity: density),
+          child: Column(
+            children: <Widget>[
+              MenuBar(
+                enabled: enabled,
+                controller: controller,
+                menus: <MenuItem>[
+                  MenuBarMenu(
+                    label: TestMenu.mainMenu1.label,
+                    onOpen: () {
+                      _openItem(TestMenu.mainMenu1);
+                    },
+                    onClose: () {
+                      _closeItem(TestMenu.mainMenu1);
+                    },
+                    menus: <MenuItem>[
+                      MenuBarItem(
+                          label: TestMenu.subMenu1.label,
```

Comment:
nit: something is off with the indentation here. should be indented by 2. Probably missing a trailing comma somewhere.